api/spec/requests/spree/api/v2/storefront/products_spec.rb

```diff
@@ -1,13 +1,146 @@
 require 'spec_helper'
 
 describe 'API V2 Storefront Products Spec', type: :request do
-  let(:product) { create(:product) }
+  let!(:products)           { create_list(:product, 5) }
+  let(:taxon)               { create(:taxon) }
+  let(:product_with_taxon)  { create(:product, taxons: [taxon]) }
+  let(:product_with_name)   { create(:product, name: 'Test Product') }
+  let(:product_with_price)  { create(:product, price: 13.44) }
+  let!(:option_type)        { create(:option_type) }
+  let!(:option_value)       { create(:option_value, option_type: option_type) }
+  let(:product_with_option) { create(:product, option_types: [option_type]) }
+  let!(:variant)            { create(:variant, product: product_with_option, option_values: [option_value]) }
+  let(:product)             { create(:product) }
+
+  shared_examples 'returns proper status' do
+    it 'returns a proper HTTP status' do
+      expect(response.status).to eq 200
+    end
+  end
+
+  describe 'products#index' do
+    context 'with no params' do
+      before { get '/api/v2/storefront/products' }
+
+      it_behaves_like 'returns proper status'
+
+      it 'returns all products' do
+        expect(json_response['data'].count).to eq Spree::Product.count
+        expect(json_response['data'].first).to have_type('product')
+      end
+    end
+
+    context 'with specified ids' do
+      before { get "/api/v2/storefront/products?ids=#{products.first.id}" }
+
+      it_behaves_like 'returns proper status'
+
+      it 'returns products with specified ids' do
+        expect(json_response['data'].count).to eq 1
+        expect(json_response['data'].first).to have_id(products.first.id.to_s)
+      end
+    end
+
+    context 'with specified price' do
```

Comment:
@VygovskySergey please update this test also to include price min / max range? Besides that, everything looks really great! Thanks!